packages/loader/src/gltf/extensions/MeshoptDecoder.ts

```diff
@@ -0,0 +1,203 @@
+// This file is part of meshoptimizer library and is distributed under the terms of MIT License.
+// Copyright (C) 2016-2022, by Arseny Kapoulkine (arseny.kapoulkine@gmail.com)
+
+import { SystemInfo } from "@galacean/engine-core";
+
+interface WorkerItem {
+  object: Worker;
+  pending: number;
+  requests: Record<string, any>;
+}
+
+const MeshoptDecoder = (function () {
```

Comment:
Carefully refactor into class and add types, etc.

Reply:
I don't think it will be benifited by refactoring into class. We will not maintain it. And if the library has been upgraded, we can hardly update the file if we refactor the file.

Reply:
Another way for maintainance is external this module like babylon.js.